app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,16 @@
+class CommentsController < ApplicationController
+  def create
+    comment_params = params.require(:comment).permit(:text, :post_id)
+    comment = Comment.new(comment_params)
+    comment.user = current_user
+    comment.post = Post.find(comment_params[:post_id])
+    respond_to do |format|
+      format.html { redirect_to user_post_path(current_user, comment.post) }
+      if comment.save
+        flash[:notice] = 'Comment created successfully'
+      else
+        flash[:alert] = 'Comment not created'
+      end
+    end
+  end
```

Comment:
- [ ]  [ OPTIONAL] THis will work fine but instead of using all in one `create` method is not good . kindly use also stong parameters  which increase security. redirection or message you can use accordingly
```ruby
def new
    @comment = Comment.new
  end

	def create
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]

    if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end
  end

  # Make it optional and put a lesson's link if students don't have the params method.  
  # Lesson link about the strong parameters: https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides.
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```